usr/lib/common/btree.c

```diff
@@ -36,37 +69,24 @@
 struct btnode *bt_get_node(struct btree *t, unsigned long node_num)
 {
     __transaction_atomic {      /* start transaction */
-        struct btnode *temp = t->top;
-        unsigned long i;
-
-        if (!node_num || node_num > t->size)
-            return NULL;
-        if (node_num == 1) {
-            temp = t->top;
-            goto done;
-        }
-
-        i = node_num;
-        while (i != 1) {
-            if (i & 1) {
-                /* If the bit is 1, traverse right */
-                temp = temp->right;
-            } else {
-                /* If the bit is 0, traverse left */
-                temp = temp->left;
-            }
-            i >>= 1;
-        }
-done:
-        return ((temp->flags & BT_FLAG_FREE) ? NULL : temp);
+        return __bt_get_node(t, node_num);
```

Comment:
well, im sure you tested it, but does this actually work: calling a function from atomic transaction w/o marking it transaction save ?

Reply:
As far as I remember a function that does not use anything that makes it transaction-unsafe, is implicitly transaction-save, and can thus be called within a transaction block.

Reply:
i also cannot reproduce the old clobbered warnings with the tm build, so it seems you fixed that one as well ;-)